src/firebase_functions/params.py

```diff
@@ -291,52 +330,68 @@ def value(self) -> str:
 class IntParam(Param[int]):
     """A parameter as a int value."""
 
+    @property
     def value(self) -> int:
         if _os.environ.get(self.name) is not None:
             return int(_os.environ[self.name])
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return int()
 
 
 @_dataclasses.dataclass(frozen=True)
 class FloatParam(Param[float]):
     """A parameter as a float value."""
 
+    @property
     def value(self) -> float:
         if _os.environ.get(self.name) is not None:
             return float(_os.environ[self.name])
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return float()
 
 
 @_dataclasses.dataclass(frozen=True)
 class BoolParam(Param[bool]):
     """A parameter as a bool value."""
 
+    @property
     def value(self) -> bool:
         env_value = _os.environ.get(self.name)
         if env_value is not None:
-            if env_value.lower() in ["true", "t", "1", "y", "yes"]:
-                return True
-            if env_value.lower() in ["false", "f", "0", "n", "no"]:
-                return False
-            raise ValueError(f"Invalid value for {self.name}: {env_value}")
+            return env_value.lower() == "true"
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return False
 
 
 @_dataclasses.dataclass(frozen=True)
 class ListParam(Param[list]):
     """A parameter as a list of strings."""
 
+    @property
     def value(self) -> list[str]:
         if _os.environ.get(self.name) is not None:
+            # If the environment variable starts with "[" and ends with "]",
+            # then assume it is a JSON array and try to parse it.
+            # (This is for Cloud Run (v2 Functions), the environment variable is a JSON array.)
+            if _os.environ[self.name].startswith("[") and _os.environ[
+                    self.name].endswith("]"):
+                try:
+                    return _json.loads(_os.environ[self.name])
+                except _json.JSONDecodeError:
+                    return []
+            # Otherwise, split the string by commas.
+            # (This is for emulator & the Firebase CLI generated .env file, the environment
+            # variable is a comma-separated list.)
```

Comment:
@taeold FYI I had to handle both scenarios here; local .env is generated as comma delimited string:

![image](https://user-images.githubusercontent.com/5347038/236097422-82c1e3d8-6fda-4857-9d7d-56973bc11241.png)

cloud run env is json array string:

![image](https://user-images.githubusercontent.com/5347038/236097483-b024225f-081b-4211-a0e6-6447b642a210.png)